README.md

```diff
@@ -1,29 +1,75 @@
-jmx4prometheus
+JMX Exporter
 =====
 
-JMX to prometheus bridge.
+JMX to Prometheus bridge.
 
-A webserver that exposes mBeans of a JMX target in JSON format suitable
-for http://github.com/prometheus.
-
-# About
-A prometheus server makes a JSON request to a the jmx4prometheus server which then
-scrapes all beans of a target JMX port on another process. The target can
-either be statically configured in the configuration or passed in by
-prometheus server as a query string parameter.
-Jmx4prometheus can return all mBeans (that are numeric) or be configured with
-whitelist/blacklist behaviour.
+A Collector that can configurably scrape and expose mBeans of a JMX target.
+This can be run as an independant HTTP server, and scrape remote JMX targets.
 
 ## Building and Running
 
 `mvn package` to build.
-See `run_jmx4prometheus_server.sh` for a sample script that runs the webserver.
+See `./run_sample_httpserver.sh` for a sample script that runs the httpserver against itself.
+
+## Configuration
+The configuration is in JSON, An example with all possible options:
+```
+{ 
+  "hostPort": "127.0.0.1:1234",
+  "rules": [
+    {"pattern": "^org.apache.cassandra.metrics<type=(\w+), name=(\w+)><>Value:",
+     "name": "cassandra_$1_$2",
+     "labels": {},
+     "help": "Cassandra metric $1 $2",
+     "type": "GAUGE",
+     "attrNameSnakeCase": false},
+  ]
+}
+```
+Name     | Description
+---------|------------
+hostPort | The host and port to connect to via remote JMX. If not specified, will talk to the local JVM.
+rules    | A list of rules to apply in order, processing stops at the first matching rule. Attributes that aren't matched aren't collected. If not specified, defaults to collection everything in the default format.
+pattern  | Regex pattern to match against each bean attribute. The pattern is not anchored. Capture groups can be used in other options. Defaults to matching everything.
+attrNameSnakeCase | Converts the attribute name to snake case, this is seen in the named matched by the pattern and the default format. For example, anAttrName to an\_attr\_name.
+name     | The metric name to set. Capture groups from the `pattern` can be used. If not specified, the default format will be used.
+labels   | A map of label name to label value pairs. Capture groups from `pattern` can be used in each. `name` must be set to use this. If not specified and the default format is not being used, no labels are set.
+help     | Help text for the metric. Capture groups from `pattern` can be used. `name` must be set to use this. Defaults to the mBean attribute decription and the full name of the attribute.
+type     | The type of the metric, can be `GAUGE` or `COUNTER`. `name` must be set to use this. Defaults to `GAUGE`.
```

Comment:
This is all reference documentation, I'll have general user documentation up in about 2 weeks with the public launch